generators/python/urllib3/templates/rest.mustache

```diff
@@ -0,0 +1,248 @@
+# coding: utf-8
+
+{{>partial_header}}
+
+import io
+import json
+import re
+import ssl
+
+import urllib3
+
+from {{packageName}}.exceptions import ApiException, ApiValueError
+
+SUPPORTED_SOCKS_PROXIES = {"socks5", "socks5h", "socks4", "socks4a"}
+RESTResponseType = urllib3.HTTPResponse
+
+
+def is_socks_proxy_url(url):
+    if url is None:
+        return False
+    split_section = url.split("://")
+    if len(split_section) < 2:
+        return False
+    else:
+        return split_section[0].lower() in SUPPORTED_SOCKS_PROXIES
+
+
+class RESTResponse(io.IOBase):
+
+    def __init__(self, resp) -> None:
+        self.response = resp
+        self.status = resp.status
+        self.reason = resp.reason
+        self.data = None
+
+    def read(self):
+        if self.data is None:
+            self.data = self.response.data
+        return self.data
+
+    def getheaders(self):
+        """Returns a dictionary of the response headers."""
+        return self.response.headers
+
+    def getheader(self, name, default=None):
+        """Returns a given response header."""
+        return self.response.headers.get(name, default)
+
+
+class RESTClientObject:
```

Comment:
I was expecting this to be handled by the generator, why do we need it in the template?

Reply:
No much changes here, just to allow default timeout configuration and avoid customer having to provide the timeout setting at each call:
- line 70: retrieve default timeout from configuration and add to pool settings
- line 149: get default timeout at each call

As for other client libraries, we're using 30s setting for each timeout value ([connect and read](https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html#urllib3.util.Timeout)), default setting should be infinite otherwise (the library seems using requests module).